platform/android/app/src/main/kotlin/coop/polypoly/polypod/features/FeatureStorage.kt

```diff
@@ -2,21 +2,49 @@ package coop.polypoly.polypod.features
 
 import FeatureManifest
 import android.content.Context
+import com.google.gson.Gson
 import coop.polypoly.core.Core
 import coop.polypoly.polypod.logging.LoggerFactory
 import java.io.File
 import java.io.FileOutputStream
 import java.util.zip.ZipFile
 
-class FeatureStorage {
-    companion object {
-        @Suppress("JAVA_CLASS_ON_COMPANION")
-        private val logger = LoggerFactory.getLogger(javaClass.enclosingClass)
+data class RawCategory(
+    val id: String,
+    val name: String,
+    val features: List<String>
+)
 
-        var activeFeatureId: String? = null
+enum class FeatureCategory {
+    yourData,
+    knowHow,
+    tools
+}
+
+data class FeatureCategoryModel(
+    val category: FeatureCategory,
+    val name: String,
+    val features: List<Feature>
+)
+
+object FeatureStorage {
+    private val logger = LoggerFactory.getLogger(FeatureStorage::class.java)
+    var activeFeatureId: String? = null
+
+    val categories: MutableList<FeatureCategoryModel> = ArrayList()
+
+    fun featureForId(id: String): Feature? {
+        for (category in categories) {
```

Comment:
`categories` needs to be set by importFeatures before calling this, so a nit and a mini-nit:
1. Nit Maybe simply put this function after those ones, since apparently alphabetical order is not required?
2. Is returning null enough for dealing with this error? Maybe check for the existence of categories and return before getting into the loop?

Reply:
1. Make sense, moved it.
2. Well, if categories is empty, the loop will not execute.